src/sql/azdata.proposed.d.ts

```diff
@@ -1790,6 +1795,72 @@ declare module 'azdata' {
 		 * The path of the parent node.
 		 */
 		parentNodePath: string;
+		/**
+		 * list of filters that are supported by the node
+		 */
+		filterDefinitions?: NodeInfoFilterDefinition[];
+	}
+
+	export interface NodeInfoFilterDefinition {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeInfoFilterDataType;
+		/**
+		 * The description of the filter property
+		 */
+		description: string;
+	}
+
+	export interface NodeInfoFilterChoiceDefinition extends NodeInfoFilterDefinition {
+		/**
+		 * The list of choices for the filter property if the type is choice
+		 */
+		choices: string[];
+	}
+
+	export interface NodeInfoFilter {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeInfoFilterDataType;
+		/**
+		 * The operator of the filter property
+		 */
+		operator: NodeInfoFilterOperators;
+		/**
+		 * The applied values of the filter property
+		 */
+		values: string[];
+	}
+
+	export enum NodeInfoFilterDataType {
```

Comment:
@Charles-Gagnon maybe we should also use const enum here so that we don't have to define the enum twice?

Reply:
For the extensibility API I've been following VS Code - they never use const enums. I've never asked them why, but I presume it's because they want the underlying values to be controlled by them and not turned into constants during compilation in case something changes (like with proposed APIs). Since this is transparent to extensions anyways (since VS Code/ADS inject the enum definition into the module) that makes sense to me. 

So while it's a bit annoying to define them twice we should just keep everything consistent here.